test cases/common/223 source set realistic example/meson.build

```diff
@@ -5,14 +5,14 @@ project('sourceset-example', 'cpp')
 ss = import('sourceset')
 kconfig = import('unstable-kconfig')
 
-zlib = dependency('zlib', version : '>=1.2.8', required: false)
+zlib = declare_dependency(compile_args: '-DZLIB=1')
 not_found = dependency('not-found', required: false)
 
 common = ss.source_set()
 specific = ss.source_set()
 
 common.add(files('main.cc'))
-common.add(when: zlib, if_true: files('zlib.cc'))
+common.add(when: zlib, if_true: [files('zlib.cc'), zlib])
```

Comment:
This change should not be necessary, the dependencies from `when` are included in `all_dependencies()`.

Reply:
yup, leftover